components/forms/Donate/Donateguild.tsx

```diff
@@ -28,25 +28,31 @@ export default function DonateGuild() {
 
   console.log(input)
 
-  const handleSubmit = async (e) => {
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault()
     setLoading(true)
 
-    const response = await fetchPostJSON('/api/checkout_sessionsguild', {
-      amount: input,
-    })
+    try {
+      const response = await fetchPostJSON('/api/checkout_sessionsguild', {
+        amount: input,
+      })
 
-    if (response.statusCode === 500) {
-      console.error(response.message)
-      return
-    }
+      if (response.statusCode === 500) {
+        console.error(response.message)
+        return
+      }
 
-    const stripe = await getStripe()
-    const { error } = await stripe.redirectToCheckout({
-      sessionId: response.id,
-    })
-    console.warn(error.message)
-    setLoading(false)
+      const stripe = await getStripe()
+      if (stripe !== null) {
+        const { error } = await stripe.redirectToCheckout({
+          sessionId: response.id,
+        })
+        console.warn(error.message)
+      }
+      setLoading(false)
+    } catch (error) {
+      console.error(error)
+    }
```

Comment:
Did you need to introduce the `try`/`catch` for the conversion?

Reply:
Yeah, this is changing how the code originally functioned. Not necessary for the typescript conversín